File: AuctionServer/AuctionServer.Domain/Entities/Payment.cs
using AuctionServer.Domain.Enums;

namespace AuctionServer.Domain.Entities;

public class Payment
{
    public Guid Id { get; set; }
    public float Amount { get; set; }
    public DateTime Date { get; set; }
    public Status Status { get; set; }
    
    public Guid AuctionItemId { get; set; }
    public AuctionItem AuctionItem { get; set; }
    
    public Guid UserId { get; set; }
    public User User { get; set; }
}

File: AuctionClient/src/types/Payment.ts
export interface Payment {
  id: string;
  amount: number;
  date: string;
  status: Status;
  auctionItemId: string;
  userId: string;
}
export interface PaymentCreation {
  amount: number;
  auctionItemId: string;
  userId: string;
}

export type Status = 1 | 2 | 3;

export const StatusText: Record<Status, string> = {
  1: "Pending",
  2: "Failed",
  3: "Completed",
};
